interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketOption;
+import java.net.UnknownHostException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.spi.SelectorProvider;
+import java.text.MessageFormat;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+@RunWith(JUnit4.class)
+public class NettyFlowControlTest {
+
+  // in bytes
+  private static final int LOW_BAND = 1024 * 1024;
+  private static final int MED_BAND = 2 * 1024 * 1024;
+  private static final int HIGH_BAND = 5 * 1024 * 1024;
+
+  // in milliseconds
+  private static final int LOW_LAT = 50;
+  private static final int MED_LAT = 100;
+  private static final int HIGH_LAT = 200;
+
+  // in bytes
+  private static final int TINY_WINDOW = 1;
+  private static final int REGULAR_WINDOW = 64 * 1024;
+  private static final int BIG_WINDOW = 1024 * 1024;
+  private static final int MAX_WINDOW = 8 * 1024 * 1024;
+
+  private static Logger logger;
+  private static TestLogHandler logHandler;
+  private static ManagedChannel channel;
+  private static InetAddress addr;
+  private static Server server;
+  private static String loopback = "127.0.0.1";
+
+  // TODO: make ports/proxy arguments
+  private final int port = 5001;
+  private final int proxyPort = 5050;
+  private final boolean useProxy = true;
+  private int testsPassed;
+  private int testsFailed;
+
+  private static final ThreadPoolExecutor executor =
+      new ThreadPoolExecutor(1, 10, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+  private static final TrafficControlProxy proxy =
+      new TrafficControlProxy(LOW_BAND, LOW_LAT, TimeUnit.MILLISECONDS);
+
+  // HandlerAccessor accessor = new HandlerAccessor();
+
+  /*
+   * Keeping this main class around for now because it's convenient to be able to run this from the
+   * command line.
+   */
+  public static void main(String[] args) throws UnknownHostException {
+    NettyFlowControlTest ft = new NettyFlowControlTest();
+    setUp();
+    ft.lowBandLowLatency();
+    ft.lowBandHighLatency();
+    ft.highBandLowLatency();
+    ft.highBandHighLatency();
+    ft.verySmallWindow();
+    shutDownTests();
+    ft.printResults();
+  }
+
+  public void printResults() {
+    System.out.println("Passed: " + testsPassed + " Failed: " + testsFailed);
```

Comment:
Prints and counts were left over from when I was running this from the command line because they  were convenient. I'll take them out though